### PR TITLE
Merge UI bug fixes into development branch

### PR DESCRIPTION
**Pull request by jiayingy**
# Fix bugs 
**File Manager**
- clear search input when changing directory 9bbb869
- fix delete button 6937146
- update folder path when changing directory 406783b

**Editor Viewer**
- Fix editor/viewer buttons view on smaller screen size

# Enhance UI
**File Manager**
- edit icon view

### CODE REVIEW

File: sashimi-webapp/src/components/editor-viewer/Content.vue
Comment (chuajiaxuan) [inline sashimi-webapp/src/components/editor-viewer/Content.vue:54]: @jiayingy Are all the "no break" cases purposely made like that? Or supposed to have the break statement?

**Review by chuajiaxuan — COMMENTED** [review]
Other than the specific comment, looks good to go! Shall wait for reply before approving! :)

File: sashimi-webapp/src/App.vue
Comment (amoshydra) [inline sashimi-webapp/src/App.vue:151]: Un-prefixed CSS are used here for common HTML tag. It will affect the entire webapp.
However, if this is intentional, then it is fine.

Reason being, in the coming version, we will put all the viewer's content inside an iframe. Therefore, this css will not be able to pollute the content in iframe anymore.

File: sashimi-webapp/src/components/file-manager/Folder.vue
Comment (amoshydra) [inline sashimi-webapp/src/components/file-manager/Folder.vue:64]: This is fascinating! 👍
Didn't know such API exist

File: sashimi-webapp/src/components/editor-viewer/Content.vue
Comment (amoshydra) [inline sashimi-webapp/src/components/editor-viewer/Content.vue]: I think we should have this `768 ` saved as a constant. It is being used in multiple places in the code.

File: sashimi-webapp/src/components/editor-viewer/Navbar.vue
Comment (amoshydra) [inline sashimi-webapp/src/components/editor-viewer/Navbar.vue]: TODO: Refactor this code in the future.

File: sashimi-webapp/src/components/file-manager/FileManager.vue
Comment (amoshydra) [inline sashimi-webapp/src/components/file-manager/FileManager.vue:109]: Download function is now implemented in logic/FileManager. After this pull request is merge, the next version should use the new API instead.

File: sashimi-webapp/src/components/file-manager/UserInputs.vue
Comment (amoshydra) [inline sashimi-webapp/src/components/file-manager/UserInputs.vue:144]: One vue-data `viewMode` is sufficient for holding this information. `iconViewMode` and `listViewMode` is not necessary. As the CSS can make use of the `viewMode` directly. e.g. 

```
div[viewMode="iconMode"] {
}
div[viewMode="listMode"] {
}
```

Using this approach, we can easily expand additional viewMode easily with just CSS stylesheet

File: sashimi-webapp/src/components/file-manager/UserInputs.vue
Comment (amoshydra) [inline sashimi-webapp/src/components/file-manager/UserInputs.vue:287]: Similar with `786px`, viewport width should be stored as a scss variable

File: sashimi-webapp/src/components/file-manager/FileManager.vue
Comment (amoshydra) [inline sashimi-webapp/src/components/file-manager/FileManager.vue:23]: "blah"

**Review by amoshydra — APPROVED** [review]

**Review by chuajiaxuan — APPROVED** [review]

File: sashimi-webapp/src/components/editor-viewer/Content.vue
Comment (amoshydra) [reply sashimi-webapp/src/components/editor-viewer/Content.vue:54]: @jiayingy it is a good practice to explicitly mention that you are using the "fall-through" behaviour in switch case. So that other developers will not mistaken this as an potential error.

**Review by amoshydra — APPROVED** [review]

**Comment by amoshydra** [comment]
@jiayingy Do review the comments here and make change in the next version

**Pull request merged**
